src/segment_reshape/map_tool/segment_reshape_tool.py

```diff
@@ -158,6 +166,26 @@ def _handle_mouse_click_event(
                 success=True,
             )
 
+    def _handle_key_event(self, key: Qt.Key) -> None:
+        if self.reshape_mode is True:
+            if key == Qt.Key_Escape or (
+                key == Qt.Key_Backspace
+                and self.new_segment_rubber_band.numberOfVertices() == 0
+            ):
+                self.new_segment_rubber_band.reset()
+                self.temporary_new_segment_rubber_band.reset()
+                self._change_to_pick_location_mode()
+                return
+            elif key in (Qt.Key_Backspace, Qt.Key_Delete):
+                if self.new_segment_rubber_band.numberOfVertices() > 1:
+                    self.new_segment_rubber_band.removeLastPoint(0, True)
+                self.temporary_new_segment_rubber_band.reset()
+                self.temporary_new_segment_rubber_band.addPoint(
+                    self.new_segment_rubber_band.getPoint(
+                        0, self.new_segment_rubber_band.numberOfVertices() - 1
+                    )
```

Comment:
this will probably return none if only one point is digitized? after the first point is clicked, temp segment is from point to cursor, and new segment is the single point. if delete is pressed line 180 removes the only point and here the new segment will be empty, ie. `getPoint(0, -1)` gets called?

should the tool return to the initial "after click, before anything is digitized" state in case delete is pressed if only point is so far present? new segment empty, temp segment from original segment start to cursor to indicate the direction

Reply:
Line 180 will be executed only if new segment has at least 2 vertices? In my opinion that guarantees that if user has ever digitized the first vertex point, it will never be removed from new segment and thus getPoint does not return None but rather redoes the selection of the first digitized vertex point every time when backspace / delete is pressed with less than 2 vertices?

Reply:
But yes, it might be more clear to completely reset new segment when pressing backspace / delete with only one vertex. I was trying to mimic "initial state" by reselecting the first point user has digitized in reshape mode to new segment but maybe we could allow the user to change the first digitized point too, at least if we are able to indicate the direction with temp segment.

I will try to fix this.